### DISCUSSION TITLE
Customize Head preflight request sent to verify Link tag

### DISCUSSION THREAD

**Discussion by hemkaran** [vercel/next.js#45777] [Help]
I am having namespaced application whose routing is handled using nginx. Nginx redirects all requests starting from `/namespace` to the NextJS application.
I am having a `<Link>` tag which simply has an href to another page in the application. Let's assume I have the following structure:
- pages/namespace/abc_page
- pages/namespace/xyz_page

On `xyz_page`, I have `<Link href="/namespace/abc_page">Go to Abc</Link>`. When I click on this link, next sends a `HEAD` request to `/en/namespace/abc_page` to check if this page exists in NextJS which results in 404 at the Nginx level as the request is starting with `/en` instead of `/namespace` and Next considers it as a normal href link and redirects the whole application to `/namespace/abc_page` which results in another server call. I am not able to enable client-side navigation in my application. Can someone help here?

**Comment by ciruz** [comment]
Hey,

I think there are some informations missing here, do you use i18n or something like that?

Maybe you have an example code / repo for us?

**Reply by hemkaran** [reply]
Yes, I am using i18n along with Next. Used this documentation, https://nextjs.org/docs/advanced-features/i18n-routing#disabling-automatic-locale-detection
I don't have any repo as I have to set up a nginx server in test environment. Let me know if you need an example code, I can create one and share.
